### PR TITLE
Implement access tracking for containingUrl

### PR DESCRIPTION
**Pull request by ntkme**
https://github.com/sass/sass/pull/3835
https://github.com/sass/dart-sass/pull/2220

### CODE REVIEW

**Comment by ntkme** [comment]
Looks like yesterday's node [20.12.2](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V20.md#20.12.2) is breaking our test on windows: https://github.com/nodejs/node/commit/69ffc6d50d

**Comment by ntkme** [comment]
@nex3 I added a commit to fix the node regression.

Regarding CVE-2024-27980:
- `sass-embedded >=1.59.2` is not affected because in production release we always launch the `dart.exe` directly in production release since this commit: https://github.com/sass/embedded-host-node/commit/308862033e00f7a28a83c3114941efb053c395f6
- `sass-embedded <1.59.2` would be broken if a windows user upgrades to the latest node. There is no security risk for `sass-embedded` itself even if a windows user does not upgrade node, because we don't have any arguments when launching `.bat`, thus there is no risk of injection.
- Test is broken because we use the `.bat` wrapper in the test.

**Comment by nex3** [comment]
Can you pull the CVE fix into a separate PR?

**Comment by ntkme** [comment]
@nex3 Created a separate PR for the CVE: https://github.com/sass/embedded-host-node/pull/286

**Review by nex3 — CHANGES_REQUESTED** [review]

File: lib/src/importer-registry.ts [hunk 1]
```diff
@@ -14,6 +14,31 @@ import {catchOr, thenOr, PromiseOr} from './utils';
...
+  }
+
+  /**
+   * Whether the `containingUrl` getter has been accessed.
+   *
+   * This is marked as public so that the importer registry can access it, but
+   * it's not part of the package's public API and should not be accessed by
+   * user code. It may be renamed or removed without warning in the future.
+   */
+  containingUrlAccessed = false;
```
[the remark this answers is not in the document]
Comment (ntkme) [reply lib/src/importer-registry.ts]: Updated. I also made the same change for ArugmentList, which is where I copied the pattern.

**Review by nex3 — APPROVED** [review]

**Pull request merged**